src/transformers/modeling_lxmert.py

```diff
@@ -761,47 +885,55 @@ def forward(
+        encoder_outputs = self.encoder(
             embedding_output,
             extended_attention_mask,
             visn_feats=visual_feats,
             visn_attention_mask=extended_visual_attention_mask,
             output_attentions=output_attentions,
         )
 
-        v_outputs, l_outputs = x_encoder_outputs[0], x_encoder_outputs[1]
-        visn_output = v_outputs[0][-1]
-        lang_output = l_outputs[0][-1]
+        visn_encoder_outputs, lang_encoder_outputs = encoder_outputs[:2]
+        all_visn_hidden_states = visn_encoder_outputs[0]
+        all_lang_hidden_states = lang_encoder_outputs[0]
+
+        all_attentions = ()
+        if output_attentions:
+            all_l_attentions = lang_encoder_outputs[1]
+            all_v_attentions = visn_encoder_outputs[1]
+            all_x_attentions = encoder_outputs[2]
+            all_attentions = (all_l_attentions, all_v_attentions, all_x_attentions)
+
+        all_hidden_states = (all_lang_hidden_states, all_visn_hidden_states) if output_hidden_states else ()
+
+        visn_output = all_visn_hidden_states[-1]
+        lang_output = all_lang_hidden_states[-1]
         pooled_output = self.pooler(lang_output)
 
         if not return_dict:
             return (
-                (lang_output, visn_output, pooled_output),
-                (v_outputs[0] if output_hidden_states else None, l_outputs[0] if output_hidden_states else None),
-                (
-                    v_outputs[1:] if output_attentions else None,
-                    l_outputs[1:] if output_attentions else None,
-                    x_encoder_outputs[2:] if output_attentions else None,
-                ),
-            )
+                lang_output,
+                visn_output,
+                pooled_output
+            ) + all_hidden_states + all_attentions
 
         return LxmertModelOutput(
-            pooled_output_x_encoder=pooled_output,
-            last_hidden_state_l=lang_output,
-            last_hidden_state_v=visn_output,
-            hidden_states_l=l_outputs[0] if output_hidden_states else None,
-            hidden_states_v=v_outputs[0] if output_hidden_states else None,
-            attentions_l_encoder=l_outputs[1:] if output_attentions else None,
-            attentions_v_encoder=v_outputs[1:] if output_attentions else None,
-            attentions_x_encoder=x_encoder_outputs[2:] if output_attentions else None,
+            pooled_output=pooled_output,
+            language_output=lang_output,
+            vision_output=visn_output,
+            language_hidden_states=all_lang_hidden_states if output_hidden_states else None,
+            vision_hidden_states=all_visn_hidden_states if output_hidden_states else None,
+            language_attentions=all_l_attentions if output_attentions else None,
+            vision_attentions=all_v_attentions if output_attentions else None,
+            cross_encoder_attentions=all_x_attentions if output_attentions else None,
```

Comment:
This is very important. It updates to more coherent returns, with a flat list of returns instead of the previous tuple of tuples.

The tuple returns must have the same value as the model outputs if cast to a tuple.

Reply:
Please note that this will potentially break your existing code bases if relying on this code.